unic/char/range/src/step.rs

```diff
@@ -0,0 +1,40 @@
+use std::char;
+
+const BEFORE_SURROGATE: char = '\u{D7FF}';
+const AFTER_SURROGATE: char = '\u{E000}';
+
+/// Step a `char` toward `char::MAX`.
+///
+/// If the given `char` is already `char::MAX`, it is returned unchanged.
+#[allow(unsafe_code)]
+pub fn forward(c: char) -> char {
+    if c == char::MAX {
```

Comment:
So, these checks result in an `open_range()` to silently create a `CharRange` with `first` > `last`. For example, if we go with `open_range('\u{0x20}', '\u{0x21}'), then `first == U+21` and `last = U+20`. Does this violate any invariants, or we don't care?

(It doesn't look like causing any issues right now, but may be with expansions of the API.)